companion/gateway/security/auth.py

```
@@ -0,0 +1,77 @@
+            status_code=401,
+            detail="Authentication required. Provide X-User-ID header."
+        )
+
+    try:
+        user = await user_service.get_user_profile(x_user_id)
+        if not user:
+            logger.warning(f"Admin endpoint accessed by non-existent user: {x_user_id}")
+            raise HTTPException(
+                status_code=404,
+                detail=f"User {x_user_id} not found"
+            )
```

Ruff (0.13.2)

Abstract `raise` to an inner function

(TRY301)